platforms.md

```diff
@@ -50,3 +50,57 @@ the `ore <platform> gc` command.
+ - `kola` works entirely on ARM based authentication, `ore` has methods for both ASM or ARM credentials.
+ - `GC` in Azure searches for Resource Groups with a prefix of `kola-cluster` in the name.
+
+## DigitalOcean
+
+ - The DO platform wraps [godo](https://github.com/digitalocean/godo).
+ - DigitalOcean has no method for uploading custom images, as a result the `ore do create-image` command does a [~~terrifying~~ special workaround](https://github.com/coreos/mantle/blob/master/cmd/ore/do/create-image.go#L117-L173) which specifies custom userdata that does the following (after which the machine is snapshotted):
+   - configure networking in the initramfs
+   - Download a custom image
+   - replaces `/root/initramfs/shutdown` with a script to:
+      - unmount the root filesystem
+      - dd the custom image over the disk
+      - reboot
+ - SSH keys are passed via both DO metadata and the userdata.
+ - UserData is passed via the DO metadata service.
+ - Instances are given the tag `mantle` which is used by `GC`.
+ - The `do-image` parameter accepts snapshot ID's as well as the channel names, if a channel is given the relevant slug `coreos-<channel>` will be passed to DO.
+
+## ESX
+
+ - The ESX platform wraps [govmomi](https://github.com/vmware/govmomi).
+ - ESX is actually a bit of a misnomer, it actually requires that the target is a vSphere instance (as it uses features not present in the free ESXi hypervisor).
+ - The general workflow for running tests on ESX is to run the `ore esx create-base` to upload an OVF image and then that name is specified to `kola` via the `esx-base-vm` parameter. This is done to only require a clone operation rather than a full re-upload of the image.
+ - Userdata is passed to the instances via the `ovfenv`. When a machine is created the API updates the `config.vAppConfig.property.guestinfo.coreos.config.data` field to be a base64 encoded userdata config.
+
+## GCE
+
+ - The GCE/gcloud platform wraps [this SDK](google.golang.org/api).
+ - Instances are tagged with `created-by:mantle` which is used when filtering instances for `GC`.
+
+## Packet
+
+ - The Packet platform wraps [packngo](https://github.com/packethost/packngo).
+ - Rather than uploading custom images the machine creation actually writes a custom iPXE script (which is uploaded to Google Storage) that sets `coreos.config.url` on the kernel command-line to point at a userdata file (which is also uploaded to Google Storage). This userdata file contains multiple systemd units & file definitions: the actual metadata is written to `/userdata`, a noop Ignition config is written to `/noop.ign` (this is used only if the given userdata is not an Ignition config). The systemd units will run `coreos-install` to install the custom image on the machine (and pass the config file).
```

Comment:
`noop.ign` is one of several implementation details probably not worth mentioning here.

Reply:
Yeah, given it only has bearing if we're passing a `cloudinit` I wasn't sure if I should expand to cover that detail or just omit the `noop.ign`

Reply:
`Rather than uploading custom images` is a bit misleading.  Custom images are fetched from a URL during the `coreos-install` process.  We don't use the Packet [custom image mechanism](https://help.packet.net/technical/infrastructure/custom-images), though.

Reply:
How about: `Custom images do not use the Packet Custom Images API but rather the machine creation...`

Reply:
Sure.